cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -94,18 +98,21 @@ public void validateProperties(Map<String, String> properties) {
     // Lower case letters and dashes allowed only.
     List<String> networkTags = conf.getNetworkTags();
     if (!networkTags.stream().allMatch(e -> NETWORK_TAGS_PATTERN.matcher(e).matches())) {
-      throw new IllegalArgumentException("Invalid Network Tags: Ensure tag length is max 63 chars"
-                                           + " and contains  lowercase letters, numbers and dashes only. ");
+      throw new InvalidConfigurationException("Invalid Network Tags: Ensure tag length is max 63 chars"
+                                           + " and contains  lowercase letters, numbers and dashes only. ",
+              ErrorGroup.CONFIGURATION, ErrorCodeProvider.NETWORK_TAGS);
     }
+
     if (networkTags.size() > 64) {
       throw new IllegalArgumentException("Exceed Max number of tags. Only Max of 64 allowed. ");
     }
 
     if (!isAutoscalingFieldsValid(conf, properties)) {
-      throw new IllegalArgumentException(
+      throw new InvalidConfigurationException(
         String.format("Invalid configs : %s, %s, %s. These are not allowed when %s is enabled ",
                       DataprocConf.WORKER_NUM_NODES, DataprocConf.SECONDARY_WORKER_NUM_NODES,
-                      DataprocConf.AUTOSCALING_POLICY , DataprocConf.PREDEFINED_AUTOSCALE_ENABLED));
+                      DataprocConf.AUTOSCALING_POLICY , DataprocConf.PREDEFINED_AUTOSCALE_ENABLED),
+              ErrorGroup.DATAPROC, ErrorCodeProvider.INVALID_CONFIGURATION);
```

Comment:
seems better to be more specific, that these are invalid autoscale configs

Reply:
how do we decide that these configuration errors are dataproc group while the others are config group?

Reply:
Yes, I was not consistent earlier. Made all InvalidConfigurationExceptions related to dataproc have the same errorgroup which is DATAPROC.

Reply:
Changed INVALID_CONFIGURATION to INVALID_AUTOSCALE_CONFIGURATION

Reply:
As discussed, instead of groups added tags. Multiple tags may be present for each exception.